serve/mlc_serve/engine/sync_engine.py

```diff
@@ -352,9 +352,8 @@ def _decode_last_output(self, state: RequestState) -> str:
         return full[len(prefix) :]
 
     def _should_stop_by_length(self, state: RequestState) -> bool:
-        # TODO: put to config
-        max_tokens = 4096
+        max_tokens = self.text_generator.model.chat_config.max_gen_len
```

Comment:
> Can we simply use the maximum context length in HF config? This also matches with OpenAI API.
I do not fully understand what kind of simplification you are proposing. OpenAI api reder to parameter in the request. It is also handled in the `_should_stop_by_length` function. The min value will be selected. We cannot have the only one. We have two parameters - one is defined by model, another is defined by the user in the request

> Note, different llama models use different variable name to encode maximum context length. 
it does not matter since we do not work with llama config parameters directly. We work with parameter which is written into chat_config and it is under our control to have the same param for any model. This param really is defined by [PR1249](https://github.com/mlc-ai/mlc-llm/pull/1249)

Reply:
> I do not fully understand what kind of simplification you are proposing. OpenAI api reder to parameter in the request. It is also handled in the _should_stop_by_length function. The min value will be selected. We cannot have the only one. We have two parameters - one is defined by model, another is defined by the user in the request

iiuc, `self.text_generator.model.chat_config.max_gen_len` here is actually equal to max context length defined in the HF config?

Reply:
> here is actually equal to max context length defined in the HF config
Not necessary. max_sequence_length can be or cannot be in the original HF model. The max_sequence_length, that you refer to, is a field in the mlc-llm build data structure. It can be initialized and really initialized by `max_position_embeddings`  from HF model.

I.e. we have
*  `max_position_embeddings` or `max_sequence_length` in original model
* `max_sequence_length` in the mlc-llm build flow that can be initialized by above parameters from the HF model or explicitly overridden by `max_seq_len` passed to build.py script
* `max_seq_len` - the parameter which is only written to the mlc_chat_config.json

If we want to have standalone deploy flow, we need to have all data together with compiled model. This can be
*  `mlc_chat_config.json` that is preferd for me, and in this case `max_seq_len` is the only parameter that we can/need to use.
* Or we can introduce the new config file for batch serve solution, that is less preferred to me. In this case we can name parameter in this new config file as `max_sequence_length`

Reply:
@elvin-n, my main concern with `mlc_chat_config.json` is that it contains too much unnecessary info for mlc_serve. For example, `ConvConfig`, `model_category`, `top_p`, `mean_genlen`, ... I think this is what @jroesch also mean in this [PR](https://github.com/octoml/mlc-llm/pull/65#pullrequestreview-1732378931). 

I worked on this quickly yesterday for the latter option. Let me send the PR soon so that we can discuss over there.